go/store/nbs/table.go

```diff
@@ -145,50 +140,21 @@ func crc(b []byte) uint32 {
 	return crc32.Update(0, crcTable, b)
 }
 
-func computeAddrDefault(data []byte) addr {
+func computeHashDefault(data []byte) hash.Hash {
 	r := sha512.Sum512(data)
-	h := addr{}
-	copy(h[:], r[:addrSize])
-	return h
+	return hash.New(r[:hash.ByteLen])
 }
 
-var computeAddr = computeAddrDefault
+var computeAddr = computeHashDefault
 
-type addr [addrSize]byte
-
-var encoding = base32.NewEncoding("0123456789abcdefghijklmnopqrstuv")
-
-func (a addr) String() string {
-	return encoding.EncodeToString(a[:])
-}
-
-func (a addr) Prefix() uint64 {
-	return binary.BigEndian.Uint64(a[:])
-}
-
-func (a addr) Suffix() []byte {
-	return a[addrPrefixSize:]
-}
-
-func parseAddr(str string) (addr, error) {
-	var h addr
-	_, err := encoding.Decode(h[:], []byte(str))
-	return h, err
-}
-
-func ValidateAddr(s string) bool {
-	_, err := encoding.DecodeString(s)
-	return err == nil
-}
-
-type addrSlice []addr
+type addrSlice []hash.Hash
```

Comment:
Nit: delete this and use hash.HashSlice instead?

Reply:
love this plan